diff -uNr nginx-1.2.6/src/event/ngx_event.c nginx/src/event/ngx_event.c
--- nginx-1.2.6/src/event/ngx_event.c	2012-09-25 02:54:28.000000000 +0800
+++ nginx/src/event/ngx_event.c	2013-03-04 13:29:20.974618760 +0800
@@ -208,7 +208,7 @@
         flags = 0;
 
     } else {
-        timer = ngx_event_find_timer();
+        timer = ngx_timer_expire_time();
         flags = NGX_UPDATE_TIME;
 
 #if (NGX_THREADS)
@@ -260,7 +260,7 @@
     }
 
     if (delta) {
-        ngx_event_expire_timers();
+        ngx_timer_process_expired();
     }
 
     ngx_log_debug1(NGX_LOG_DEBUG_EVENT, cycle->log, 0,
diff -uNr nginx-1.2.6/src/event/ngx_event.h nginx/src/event/ngx_event.h
--- nginx-1.2.6/src/event/ngx_event.h	2012-06-04 19:58:12.000000000 +0800
+++ nginx/src/event/ngx_event.h	2013-03-04 13:30:51.782614712 +0800
@@ -448,10 +448,6 @@
 #define ngx_add_conn         ngx_event_actions.add_conn
 #define ngx_del_conn         ngx_event_actions.del_conn
 
-#define ngx_add_timer        ngx_event_add_timer
-#define ngx_del_timer        ngx_event_del_timer
-
-
 extern ngx_os_io_t  ngx_io;
 
 #define ngx_recv             ngx_io.recv
@@ -555,7 +551,7 @@
 #define ngx_event_ident(p)  ((ngx_connection_t *) (p))->fd
 
 
-#include <ngx_event_timer.h>
+#include <ngx_timer_module.h>
 #include <ngx_event_posted.h>
 #include <ngx_event_busy_lock.h>
 
diff -uNr nginx-1.2.6/src/os/unix/ngx_process_cycle.c nginx/src/os/unix/ngx_process_cycle.c
--- nginx-1.2.6/src/os/unix/ngx_process_cycle.c	2012-12-11 01:46:51.000000000 +0800
+++ nginx/src/os/unix/ngx_process_cycle.c	2013-03-04 13:33:09.294608581 +0800
@@ -794,7 +794,7 @@
                 }
             }
 
-            if (ngx_event_timer_rbtree.root == ngx_event_timer_rbtree.sentinel)
+            if (ngx_timer_empty())
             {
                 ngx_log_error(NGX_LOG_NOTICE, cycle->log, 0, "exiting");
 
